packages/cli-kit/src/node/cli.ts

```diff
@@ -52,4 +54,50 @@ export async function runCreateCLI(options: RunCLIOptions) {
   await runCLI(options)
 }
 
+export async function replaceGlobalCLIWithLocal(filepath: string): Promise<boolean> {
+  // Temporary flag while we test out this feature and ensure it won't break anything!
+  if (!isTruthy(process.env[constants.environmentVariables.enableCliRedirect])) return false
+  if (isTruthy(process.env[constants.environmentVariables.skipCliRedirect])) return false
+  if (process.env.npm_config_user_agent) return false
+
+  const cliPackage = await localCliPackage()
+  if (!cliPackage) return false
+
+  const correctExecutablePath = join(cliPackage.path, cliPackage.bin.shopify)
+  if (correctExecutablePath === filepath) return false
+  try {
+    const env = {}
+    await exec(correctExecutablePath, process.argv.slice(2, process.argv.length), {
+      stdio: 'inherit',
+      env: {[constants.environmentVariables.skipCliRedirect]: '1'},
+    })
+    // eslint-disable-next-line no-catch-all/no-catch-all, @typescript-eslint/no-explicit-any
+  } catch (processError: any) {
+    process.exit(processError.exitCode)
+  }
+  return true
+}
+
+interface CliPackageInfo {
+  path: string
+  bin: {shopify: string}
+}
+
+async function localCliPackage(): Promise<CliPackageInfo | undefined> {
+  let npmListOutput = ''
+  try {
+    npmListOutput = await captureOutput('npm', ['list', '@shopify/cli', '--json', '-l'])
+    // eslint-disable-next-line no-catch-all/no-catch-all
+  } catch (err) {
+    return
+  }
+  const localShopifyCLI = JSON.parse(npmListOutput)
```

Comment:
I'd include this one in the `try {`. Just in case the output contains additional characters that make the JSON decoding fail.